### PR TITLE
Deploy tweaks (pre work for CD)

### PR DESCRIPTION
**Pull request by timothybest-usds**
## Related Issue or Background Info

Precursor to deploying `prod` on merge to `main`

## Changes Proposed

- fail `training`, `demo` and `prod` if `stg` and `dev` are not up to date
- remove pentest deploying on release. the pentest for ATO is over and this env doesn't need to be kept up to date
- deploy `stg` on merge along with `dev`. this removes the need to uncheck a box to deploy
- update deploy and rollback docs
- enforce concurrency on dev (https://usds.slack.com/archives/C01LTSNKEPP/p1627409378492600)

## Additional Information

- Epic: https://github.com/CDCgov/prime-simplereport/issues/1774

## Checklist for Author and Reviewer
- [ ] open to suggestions on how to test other than mergeing to `main` and testing a deploy

### UI
- [n/a] Any changes to the UI/UX are approved by design 
- [n/a] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [n/a ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
- [ n/a] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [n/a ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [n/a ] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [n/a] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

File: .github/workflows/deployPentest.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ name: Deploy Pentest
 
 on:
   workflow_dispatch:
-  release:
-    types: [released]
```
Comment (bmaden-usds) [inline .github/workflows/deployPentest.yml:6]: Woohoo, no email about failed pentest deployment after release : )

**Review by bmaden-usds — APPROVED** [review]
LGTM!

File: README.md [hunk 2]
```diff
@@ -353,31 +353,24 @@ Navigate to [New Release Form](https://github.com/CDCgov/prime-simplereport/rele
...
-5. Verify the changes are live in `stg` by ensuring the deployed commit hash matches the commit hash on the release and the deployed release tag matches. This is done by going to `/app/static/commit.txt` and `/api/actuator/info`
-6. Return to the release page and select "Edit release"
-7. Un-check the "This is a pre-release" checkbox and click "Update release" (this will trigger the release to other environments)
-8. Verify that the changes are live in `prod`, `demo` and `training`.
+3. Add a release title summarizing the changes
+4. If applicable describe some of the changes in detail in the description
+5. Click publish release
+6. Verify that the changes are live in `prod`, `demo` and `training`.
 
 ### Revert to a Previous Release
```
Comment (jdorothy) [inline README.md:366]: We should also mention the rollback workflow here, if you just need an immediate revert to the previous release.

**Review by jdorothy — APPROVED** [review]
LGTM other than one comment and a nitpick

File: README.md [hunk 3]
```diff
@@ -353,31 +353,24 @@ Navigate to [New Release Form](https://github.com/CDCgov/prime-simplereport/rele
 ![full-dialog](https://user-images.githubusercontent.com/28784751/121424756-b31bd380-c93f-11eb-987d-38934f0570ae.png)
 
 1. <img align="right" width="517" alt="select-commit" src="https://user-images.githubusercontent.com/28784751/121423065-df365500-c93d-11eb-9b95-a63130d602e6.png">
-   Select the commit you want to release. This is likely to be the last commit on `main`, but select
-   the commit explicitly so that you do not accidentally release changes that somebody else is in the
+   Select the last commit on `main`. Selecting
```
Comment (jdorothy) [inline README.md:356]: Looks like the formatting got a little chopped up?

Comment (timothybest-usds) [reply README.md:356]: this is intentional. We should no longer be cutting releases at anything other then the latest commit

**Pull request merged**